core/src/main/scala/kafka/log/Log.scala

```diff
@@ -717,16 +746,20 @@ class Log(@volatile var dir: File,
     deleteOldSegments(shouldDelete)
   }
 
+  /*
+   * The log's next offset may be smaller than log start offset when the log is purged upon PurgeRequest
+   * We don't need to call this method periodically using scheduler since it is called right after broker receives PurgeRequest
+   */
+  private def deleteLogStartOffsetBreachedSegments() : Int = {
+    // keep active segment to avoid frequent log rolling due to user's PurgeRequest
+    deleteOldSegments(segment => segment.nextOffset <= logStartOffset && segment.baseOffset < activeSegment.baseOffset)
```

Comment:
Sure, the typo is fixed.

In the current implementation, if user seeks to earliest offset, he/she will seek to the `logStartOffset` of the leader regardless of what is the low watermark. User will not seek to LW since LW is not even exposed to the user. Is there any concern with this approach?